<a href="https://colab.research.google.com/github/damianmtz98/Aprendizaje-Automatico-MCD/blob/main/TareaSesion2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# ***Tarea en clase***
- Cargar conjunto de datos mediante pandas.
- Especificar el tipo de datos de cada columna.
- Si es necesario, asignar una columna como índice.

In [1]:
import pandas as pd
from IPython.display import Markdown

In [11]:
hpdatos = pd.read_csv('heart_prediction.csv')
df_hpdatos = pd.DataFrame(datospd)
#Importamos nuestra base de datos

In [12]:
df_hpdatos

id  Age  Sex  Chest pain type   BP  Cholesterol  FBS over 120  \
0      0   70    1                4  130          322             0   
1      1   67    0                3  115          564             0   
2      2   57    1                2  124          261             0   
3      3   64    1                4  128          263             0   
4      4   74    0                2  120          269             0   
..   ...  ...  ...              ...  ...          ...           ...   
265  265   52    1                3  172          199             1   
266  266   44    1                2  120          263             0   
267  267   56    0                2  140          294             0   
268  268   57    1                4  140          192             0   
269  269   67    1                4  160          286             0   

     EKG results  Max HR  Exercise angina  ST depression  Slope of ST  \
0              2     109                0            2.4            2   
1              2     160                0            1.6            2   
2              0     141                0            0.3            1   
3              0     105                1            0.2            2   
4              2     121                1            0.2            1   
..           ...     ...              ...            ...          ...   
265            0     162                0            0.5            1   
266            0     173                0            0.0            1   
267            2     153                0            1.3            2   
268            0     148                0            0.4            2   
269            2     108                1            1.5            2   

     Number of vessels fluro  Thallium Heart Disease  
0                          3         3      Presence  
1                          0         7       Absence  
2                          0         7      Presence  
3                          1         7       Absence  
4                          1         3       Absence  
..                       ...       ...           ...  
265                        0         7       Absence  
266                        0         7       Absence  
267                        0         3       Absence  
268                        0         6       Absence  
269                        3         3      Presence  

[270 rows x 15 columns]

# Columnas
- **Age**: The age of the patient. *int*
- **Sex**: The gender of the patient. 0 = M, 1= H.  *int*
- **Chest pain type:**	The type of chest pain experienced by the patient. Dolor 1-4 siendo el 4 mas fuerte. *int*
- **BP:**	The blood pressure level of the patient. *int* 
- **Cholesterol:** The cholesterol level of the patient. *int*
- **FBS over 120:**	The fasting blood sugar test results over 120 mg/dl. *int*
- **EKG results:**	The electrocardiogram results of the patient. 0 al 2, siendo 2 malo. *int*
- **Max HR:**	The maximum heart rate levels achieved during exercise testing. *int*
- **Exercise angina:**	The angina experienced during exercise testing. 0 - 1, siendo 1 malo. *int*
- **ST depression:**	The ST depression on an Electrocardiogram. *int*
- **Slope of ST:**	The slope of ST segment electrocardiogram readings. *float* 
- **Number of vessels fluro:**	The amount vessels seen in Fluoroscopy images. *int*
- **Thallium:**	The Thallium Stress test findings. *int*
- **Heart Disease:**	Whether or not the patient has been diagnosed with Heart Disease. *int*

# ***Tarea***
- Crear, modificar o eliminar columnas. 
- Filtrar registros.
- Crear subconjuntos de datos.
- Crear grupos de datos.

In [13]:
df_hpdatos = df_hpdatos.copy()

In [15]:
# Eliminamos nuestra columna numero 15
df_hpdatos = df_hpdatos.drop(columns = 'Exercise angina' )
df_hpdatos.head(5)

id  Age  Sex  Chest pain type   BP  Cholesterol  FBS over 120  EKG results  \
0   0   70    1                4  130          322             0            2   
1   1   67    0                3  115          564             0            2   
2   2   57    1                2  124          261             0            0   
3   3   64    1                4  128          263             0            0   
4   4   74    0                2  120          269             0            2   

   Max HR  ST depression  Slope of ST  Number of vessels fluro  Thallium  \
0     109            2.4            2                        3         3   
1     160            1.6            2                        0         7   
2     141            0.3            1                        0         7   
3     105            0.2            2                        1         7   
4     121            0.2            1                        1         3   

  Heart Disease  
0      Presence  
1       Absence  
2      Presence  
3       Absence  
4       Absence

In [22]:
# Filtramos la edad que sean mayor a 70 con la frecuencia cardiaca de mayor de 140.
ageheart = df_hpdatos[(df_hpdatos['Age'] > 65 ) & (df_hpdatos['Max HR'] > 100 ) & (df_hpdatos ['BP'] > 130)]
df_ageheart = pd.DataFrame(ageheart)
df_ageheart

id  Age  Sex  Chest pain type   BP  Cholesterol  FBS over 120  \
31    31   66    1                4  160          228             0   
48    48   66    1                2  160          246             0   
72    72   70    1                2  156          245             0   
73    73   76    0                3  140          197             0   
118  118   66    0                1  150          226             0   
136  136   67    0                3  152          277             0   
159  159   66    0                4  178          228             1   
170  170   69    1                1  160          234             1   
171  171   69    1                3  140          254             0   
189  189   70    1                3  160          269             0   
191  191   70    1                4  145          174             0   
198  198   69    0                1  140          239             0   
240  240   68    1                3  180          274             1   
255  255   71    0                2  160          302             0   
269  269   67    1                4  160          286             0   

     EKG results  Max HR  ST depression  Slope of ST  Number of vessels fluro  \
31             2     138            2.3            1                        0   
48             0     120            0.0            2                        3   
72             2     143            0.0            1                        0   
73             1     116            1.1            2                        0   
118            0     114            2.6            3                        0   
136            0     172            0.0            1                        1   
159            0     165            1.0            2                        2   
170            2     131            0.1            2                        1   
171            2     146            2.0            2                        3   
189            0     112            2.9            2                        1   
191            0     125            2.6            3                        0   
198            0     151            1.8            1                        2   
240            2     150            1.6            2                        0   
255            0     162            0.4            1                        2   
269            2     108            1.5            2                        3   

     Thallium Heart Disease  
31          6       Absence  
48          6      Presence  
72          3       Absence  
73          3       Absence  
118         3       Absence  
136         3       Absence  
159         7      Presence  
170         3       Absence  
171         7      Presence  
189         7      Presence  
191         7      Presence  
198         3       Absence  
240         7      Presence  
255         3       Absence  
269         3      Presence

In [32]:
# Creamos un subconjunto de datos
# Donde tendremos la edad, presion arterial, resultados del electrocardiograma y las enfermedades del corazon.
subcon1 = df_ageheart[['Age', 'BP', 'Max HR', 'Sex']]
df_subcon1 = pd.DataFrame(subcon1)
df_subcon1

Age   BP  Max HR  Sex
31    66  160     138    1
48    66  160     120    1
72    70  156     143    1
73    76  140     116    0
118   66  150     114    0
136   67  152     172    0
159   66  178     165    0
170   69  160     131    1
171   69  140     146    1
189   70  160     112    1
191   70  145     125    1
198   69  140     151    0
240   68  180     150    1
255   71  160     162    0
269   67  160     108    1

In [36]:
# Creamos este grupo donde buscamos los hombres que sean mas vulnerables arriba de 65 años.
gpo1 = df_subcon1.groupby(['Age'])[['Sex']].sum()
gpo1 

Sex
Age     
66     2
67     1
68     1
69     2
70     3
71     0
76     0